conans/model/conan_file.py

```diff
@@ -250,7 +251,9 @@ def _run():
             return tools.run_in_windows_bash(self, bashcmd=command, cwd=cwd, subsystem=subsystem,
                                              msys_mingw=msys_mingw, with_login=with_login)
         if run_environment:
-            with tools.run_environment(self):
+            # When using_build_profile the required environment is already applied through 'conanfile.env'
+            # in the contextmanager 'get_env_context_manager'
```

Comment:
Yes, as long as they are defined in ``self.deps_env_info`` and get values from dependencies, then this looks ok

Reply:
This is guaranteed in the lines from the PR https://github.com/conan-io/conan/pull/5592 where the `env_info` takes the values from the `cpp_info` fields of its build_requires-build.

```python
    # Add cpp_info.bin_paths/lib_paths to env_info (it is needed for runtime)
    env_info.DYLD_LIBRARY_PATH.extend(n.conanfile.cpp_info.lib_paths)
    env_info.DYLD_LIBRARY_PATH.extend(n.conanfile.cpp_info.framework_paths)
    env_info.LD_LIBRARY_PATH.extend(n.conanfile.cpp_info.lib_paths)
    env_info.PATH.extend(n.conanfile.cpp_info.bin_paths)
```

What we are missing here is the `env_info` of the plain requires and the build_requires-host